docs/recipes/maintenance-releases.md

```diff
@@ -0,0 +1,156 @@
+# Publishing maintenance releases
+
+This recipe will walk you through a simple example that uses Git branches and distribution channels to publish fixes and features for old versions of a package.
+
+This example uses the **semantic-release** default configuration:
+- [branches](../usage/configuration.md#branches): `['+([1-9])?(.{+([1-9]),x}).x', 'master', 'next', 'next-major', {name: 'beta', prerelease: true}, {name: 'alpha', prerelease: true}]`
+- [plugins](../usage/configuration.md#plugins): `['@semantic-release/commit-analyzer', '@semantic-release/release-notes-generator', '@semantic-release/npm', '@semantic-release/github']`
+
+## Initial release
+
+We'll start by making the first commit of the project, with the code for the initial release and the message `feat: initial commit`. When pushing that commit, on `master` **semantic-release** will release the version `1.0.0` and make it available on the default distribution channel which is the dist-tag `@latest` for npm.
+
+The Git history of the repository is:
+
+```
+* feat: initial commit # => v1.0.0 on @latest
+```
+
+## Releasing a breaking change
+
+We now decide to drop Node.js 6 support for our package, and require Node.js 8 or higher, which is a breaking change.  
+
+We commit that change with the message `feat: drop Node.js 6 support \n\n BREAKING CHANGE: Node.js >= 8 required` to `master`. When pushing that commit, **semantic-release** will release the version `2.0.0` on the dist-tag `@latest`.
+
+The Git history of the repository is now:
+
+```
+* feat: initial commit # => v1.0.0 on @latest
+* feat: drop Node.js 6 support \n\n BREAKING CHANGE: Node.js >= 8 required # => v2.0.0 on @latest
+```
+
+## Releasing a feature for version 1.x users
+
+One of our users request a new feature, however they cannot migrate to Node.js 8 or higher due to corporate policies.
+
+If we were to push that feature on `master` and release it, the new version would require Node.js 8 or higher as the release would also contains the commit `feat: drop Node.js 6 support \n\n BREAKING CHANGE: Node.js >= 8 required`.
+
+Instead, we create the branch `1.x` from the tag `v1.0.0` with the command `git checkout -b 1.x v1.0.0` and we commit that feature with the message `feat: a feature` to the branch `1.x`. When pushing that commit, **semantic-release** will release the version `1.1.0` on the dist-tag `@release-1.x` so users who can't migrate to Node.js 8 or higher can benefit from it.
```

Comment:
That branch would not be used until one would decide to make a commit there. So that would create potentially a lot of branches that doesn't bring any value (they would functionally to the version tag).

Also creating branches might trigger a bunch of stuff on repos (CI etc...).

Finally those branches being permanent should be created as protected branches, which we cannot do from the core with only the `git` command. Maybe I should add that in the recipe.

So it might be better to users do that when they need it.